apps/sms/services/js/activity/client.js

```diff
@@ -0,0 +1,85 @@
+/* global EventDispatcher,
+          bridge
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Reference to active bridge client instance.
+   * @type {Client}
+   */
+  var client = null;
+
+  /**
+   * Indicates if there is pending activity request.
+   * @type {boolean}
+   */
+  var hasPendingRequest = false;
+
+  var ActivityClient = {
+    /**
+     * Initialized activity service client bridge.
+     */
+    init() {
+      client = bridge.client(SERVICE_NAME);
+
+      client.on('activity-request', (request) => {
+        hasPendingRequest = true;
```

Comment:
Well, you usually check arguments and throw errors in methods -- I see it's the same here :) 

So if hasPendingRequest is already true, console.error(...), and call postError (I think postError automatically close the activity?). And add a comment to explain that it should not happen because our activities are inline